assignments/life_expectancy/cleaning.py

```diff
@@ -0,0 +1,105 @@
+        raise ValueError ("The fuction requires a DataFrame as an argument.") 
+
+
+    #Obtains the values from the first column - main objetive here is to get the country info
+    df1 = df.iloc[:, 0]
+    df1 = df1.str.split(',', n=4, expand=True)
+    #rename columns
+    new_columns = ["unit","sex","age","region"]
+    df1 = df1.set_axis(new_columns, axis=1)
+
+    #Obtain the year and values information of each country
+    df2 = df.iloc[:,1:]
+    #Remove the dots from the columns
+    df2 = df2.replace(re.compile(r"\s*:\s*"),"")
+    #Remove the letters from the columns
+    df2 = df2.replace({r'([a-zA-Z]*)':""},regex=True)
+
+    #Dataframe ready to unpivot
+    df = pd.concat([df1,df2],axis=1)
+    #Unpivot the data from the wide format to a load format
+    table_col_names = ["unit", "sex", "age", "region"]
+    df_unpivot = pd.melt(df,
+                        id_vars=table_col_names,
+                        value_vars=[i for i in df.columns if i not in table_col_names]
+                        )
+
+    # Final dataset
+    new_columns = ["unit","sex","age","region","year","value"]
+    df = df_unpivot.set_axis(new_columns, axis=1)
+
+    #Convert data types and Clean nan
+    df['year'] = df['year'].astype('int')
+    df['value'] = pd.to_numeric(df['value'])
+    df = df.dropna(subset=['value'])
+
+    #Filer the final dataset
+    df = df[df['region'] == region]
+
+    return df
+
+def save_data(df: pd.DataFrame, output_path: str|Path):
+    """Fuction that save the data into the expercted folder.
+    :param df: file that was retrieved in save_data() function. Cleaned information.
+    :param output_path: local where the file is located. Search in __init__.py to find OUTPUT_DIR
+    """
+    if df is None: 
+        raise ValueError ("The fuction requires a DataFrame as an argument.") 
+    if output_path is None: 
+        raise ValueError ("The fuction requires a path to save the file.") 
+
+     #Export that file into the folder
+    df.to_csv(output_path, index=False)
+    print(f"Finish data cleaning The file was saved as csv in:\n{output_path}\n")
+
+
+
+def main(): # pragma: no cover
+    """
+    Parser setup + call three functions defined above
+    """
+    parser = argparse.ArgumentParser()
+    parser.prog = 'cleaning.py'
+    parser.description = "This is where the command-line utility's description goes."
+    parser.epilog = "This is where the command-line utility's epilog goes."
+    parser.add_argument('-i', default = "/workspaces/CF_Faast_Foundations/assignments/life_expectancy/data/eu_life_expectancy_raw.tsv", help="You need to put here the path of the input file")
+    parser.add_argument('-r', default= 'PT', help="Filter for the region you want to select.")
+    parser.add_argument('-o', default = '/workspaces/CF_Faast_Foundations/assignments/life_expectancy/data/pt_life_expectancy.csv', help="You need to put here the path where you want to write the output file")
+    args = parser.parse_args()
```

Comment:
Where you can check if all the input parameters are given, raising an error by e.g.:

```
    # Check if all required arguments are provided
    required_args = ["i", "o"]
    for arg in required_args:
        if arg not in args:
            raise ValueError(f"Missing argument: '{arg}'")
```

Reply:
Suggestion that will not be attended because there are already default parser values.